### PR TITLE
Modernize CI 

### LINKED ISSUE #100: Modernize CI
- [ ] use pre-commit CI with styler
- [ ] build artifact of gh-pages in PRs
- [ ] deploy gh-pages only when merging into master
- [ ] pre-commit for roxygenize (?)

EDIT: actually, the current way of deploying github pages is already quite smart. It deploys each PR in a subfolder as a preview. This has the limitation that this doesn't work for PRs from a fork, so I should switch to artifacts nonetheless

### PR DESCRIPTION
**Pull request by grst**
Close #100 

 - [x] add pre-commit hooks
 - [x] roxygenize automatically
   - [x] commit as `omnideconv-bot`
 - [x] update R CMD test scripts to use r-lib/actions
 - [x] deploy gh pages to netlify on PR 
   - [x] use auth token from omnideconv-bot
 - [x] `covr`
 - [x] fix TODOs
 - [ ] use forks of MCP Counter packages to resolve issues with pak: 
    ```
    # TODO workaround until https://github.com/r-lib/pak/issues/385 is resolved
    grst/MCPcounter
    # TODO workaround until https://github.com/cit-bioinfo/mMCP-counter/pull/4 is merged
    grst/mMCP-counter
    ```

### CODE REVIEW

**Comment by grst** [comment]
Need to wait for r-lib/pak#397 and r-lib/pak#398, then decide how to move forward.

File: R/datasets.R [hunk 1]
```diff
@@ -9,3 +9,10 @@
 #'
 #' @source Racle et al. (2017), eLIFE, https://doi.org/10.7554/eLife.26476.029
 "dataset_racle"
+
+#' RNA-seq dataset with FACS measurements
+#'
+#' TODO
```
Comment (grst) [inline R/datasets.R:15]: @LorenzoMerotto could you please add some details about the dataset here?

**Review by grst — COMMENTED** [review]
Hi @LorenzoMerotto, 

I finally got the new CI working. There are a couple of novelties
 * they are now based on the latest template from [r-lib/actions](https://github.com/r-lib/actions). This made the script for `R CMD test` significantly shorter and faster. 
 * I set-up pre-commit to enforce some code-style (autoformatting of R, yml and markdown files). Pre-commit can be set-up locally (`pip install pre-commit; pre-commit install` in the repo) to run checks on every commit. If it is not set-up locally, the CI will run the checks and automatically add a commit with fixes where possible
 * If not ran locally, `roxygenize` will be executed by the CI and a commit added with the updated documentation
 * I added coverage tests with `codecov` as suggested by @federicomarini 
 * Documentation is built on each pull request, and a temporary "netlify" link generated to preview the page (see and example comment above) 

To make the CI pass, I had to fix a couple of issues in the documentation. Apparently I had disabled errors on check warnings before, otherwise they would have appeared earlier. 
For instance, the petitprez dataset doesn't have any documentation, but it is mandated by `R CMD check` that all datasets are documented. Would be great if you could add a couple of lines there. 

Other than that, please take a look and LMK if you have any questions how this works and feel free to merge if you have no concerns. In a next step I will add documentation to the developer docs (#99) how everything was set-up.

Once this is up and running here, I'll port the new CI to the other omnideconv repos.

File: R/datasets.R (hunk 1 above)
Comment (LorenzoMerotto) [reply R/datasets.R]: Did it!

**Pull request merged**